core/server/master/src/test/java/alluxio/master/meta/MetaDailyBackupTest.java

```diff
@@ -0,0 +1,113 @@
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.ConfigurationRule;
+import alluxio.PropertyKey;
+import alluxio.master.BackupManager;
+import alluxio.underfs.UfsFileStatus;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.util.CommonUtils;
+import alluxio.util.executor.ControllableScheduler;
+import alluxio.util.io.PathUtils;
+import alluxio.wire.BackupResponse;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.Closeable;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests the {@link MetaDailyBackup}.
+ */
+public class MetaDailyBackupTest {
+  private MetaMaster mMetaMaster;
+  private ControllableScheduler mScheduler;
+  private UnderFileSystem mUfs;
+  private Random mRandom;
+
+  @Test
+  public void test() throws Exception {
+    try (Closeable c =
+        new ConfigurationRule(ImmutableMap.of(PropertyKey.MASTER_DAILY_BACKUP_ENABLED, "true",
+        PropertyKey.MASTER_DAILY_BACKUP_FILES_RETAINED, "1")).toResource()) {
+      MetaDailyBackup dailyBackup = new MetaDailyBackup(mMetaMaster, mScheduler, mUfs);
+      dailyBackup.start();
+
+      // Trigger the runnable within 3 days
+      mScheduler.jumpToFuture(3, TimeUnit.DAYS);
+
+      verify(mMetaMaster, times(3)).backup(any());
+      verify(mUfs, times(3)).listStatus(any());
+      verify(mUfs, times(6)).deleteFile(any());
+    }
+  }
+
+  @Before
+  public void before() throws Exception {
+    mRandom = new Random();
+    String backupDir = Configuration.get(PropertyKey.MASTER_BACKUP_DIRECTORY);
+
+    mMetaMaster = Mockito.mock(MetaMaster.class);
+    when(mMetaMaster.backup(any())).thenReturn(new BackupResponse(new AlluxioURI(
+        PathUtils.concatPath(backupDir, generateBackupFileName())), "localhost"));
+
+    mUfs = Mockito.mock(UnderFileSystem.class);
+    when(mUfs.getUnderFSType()).thenReturn("local");
+    when(mUfs.listStatus(backupDir)).thenReturn(getUfsStatuses());
+    when(mUfs.deleteFile(any())).thenReturn(true);
+
+    mScheduler = new ControllableScheduler();
+  }
+
+  /**
+   * @return the random generated ufs file statuses
+   */
+  private UfsStatus[] getUfsStatuses() {
```

Comment:
"get" suggests that the method will return the same thing every time. How about `generateUfsStatuses`?

Reply:
Changed, thanks